### PR TITLE
fix(form-v2) : add condition to change to plain background when form empty

### PR DESCRIPTION
**Pull request by tohhengxing**
## Problem

Empty form shows white box instead of plain background

Closes [#4315]

## Solution

use conditional to set background for Box component

**Breaking Changes** 

- [x] No - this PR is backwards compatible  

## Before & After Screenshots

**BEFORE**:

look at issue #4315 

**AFTER**:

<details>
<summary>empty form</summary>

image
</details>

<details>
<summary>non empty form</summary>

image
</details>

### CODE REVIEW

File: frontend/src/features/public-form/components/FormFields/FormFields.tsx [hunk 1]
```diff
@@ -104,7 +104,11 @@ export const FormFields = ({
   return (
     <FormProvider {...formMethods}>
       <form onSubmit={formMethods.handleSubmit(onSubmit)} noValidate>
-        <Box bg="white" py="2.5rem" px={{ base: '1rem', md: '2.5rem' }}>
+        <Box
```
Comment (timotheeg) [inline frontend/src/features/public-form/components/FormFields/FormFields.tsx]: Should the box even be there at all if `formFields` is empty @karrui ?

**Comment by tohhengxing** [comment]
for the failing CI unit test, I am not sure why that's the case since the test file runs fine on my local

<details>
<summary>test pass</summary>

image
</details>

**Comment by timotheeg** [comment]
Don't worry about it @tohhengxing, this is an issue with out github setup and chromatic integration where the chromatic check cannot run for PRs coming from external forks.

**Comment by karrui** [comment]
I reran the flakey test hehe

File: frontend/src/features/public-form/components/FormFields/FormFields.tsx (hunk 1 above)
Comment (karrui) [reply frontend/src/features/public-form/components/FormFields/FormFields.tsx]: will just merge in first, let pearly decide when we re-QA

File: frontend/src/features/public-form/components/FormFields/FormFields.tsx [hunk 2]
```diff
@@ -104,7 +104,11 @@ export const FormFields = ({
...
+          bg={isEmpty(formFields) ? 'theme-grey' : 'white'}
```
Comment (karrui) [inline frontend/src/features/public-form/components/FormFields/FormFields.tsx]: ah @timotheeg i got what you mean. We can probably remove this Box if there are no fields at all. 

supernit/perf: should we also use a length check over the `isEmpty` function to determine emptiness?

Comment (timotheeg) [reply frontend/src/features/public-form/components/FormFields/FormFields.tsx]: > supernit/perf: should we also use a length check over the isEmpty function to determine emptiness?

Yes, let's do that! :D

Comment (tohhengxing) [reply frontend/src/features/public-form/components/FormFields/FormFields.tsx]: done

<details>
<summary>no box</summary>

image
</details>

Comment (tohhengxing) [reply frontend/src/features/public-form/components/FormFields/FormFields.tsx]: also, i noticed this ui inconsistency, is this intended? The opengov logo is touching the rest of the links bar at the footer
image

File: frontend/src/features/public-form/components/FormFields/FormFields.tsx [hunk 3]
```diff
@@ -104,24 +104,27 @@ export const FormFields = ({
...
-            <VisibleFormFields
-              colorTheme={colorTheme}
-              control={formMethods.control}
-              formFields={augmentedFormFields}
-              formLogics={formLogics}
-              fieldPrefillMap={fieldPrefillMap}
-            />
-          </Stack>
-        </Box>
+        {formFields?.length !== 0 && (
```
Comment (timotheeg) [inline frontend/src/features/public-form/components/FormFields/FormFields.tsx]: /nit: falsy checks suffices
```suggestion
        {formFields?.length && (
```

**Review by timotheeg — APPROVED** [review]

File: frontend/src/features/public-form/components/FormFields/FormFields.tsx (hunk 2 above)
Comment (karrui) [reply frontend/src/features/public-form/components/FormFields/FormFields.tsx]: thanks, will add an issue for it to be fixed in future PRs!

**Pull request merged**